### ISSUE TITLE
Access other layer ressources

### ISSUE THREAD

**Issue by tchaton** [keras-team/keras#9380]
Hello everyone, I have a tricky programming question. I want to implement an exponentially moving average of centroids on hidden layers inputs activations of an AutoEncoder to maximize ICC over several class in keras
(adding a loss penalizing this term).
It would require to know the final output of the classification for calculating the centroids. I could do it by adding the output of this layer as an extra output of neural network. But it would duplicate data. And it won t be nice to work with in the case if I haven t to apply this on any hidden layers.
Do you have any ideas how to do so ? Thank you

**Comment by harshini-gadige** [comment]
Hi

Since this is not a bug/feature request, request you to post these kind of questions in any of the following.

https://keras.io/
https://keras.io/#support
https://gitter.im/Keras-io/Lobby

Thanks!
